### DISCUSSION TITLE
Following build instructions, failure of skshaderc command

### DISCUSSION THREAD

**Discussion by paulmelis** [StereoKit/StereoKit#390] [Q&A]
I'm following https://github.com/StereoKit/StereoKit/blob/master/BUILDING.md#i-want-to-modify-code-windows from a clean repo (`develop` branch) to get SK to build on Windows with Visual Studio 17.1.2. I think I have all prerequisites met and alter the solution as described. But the build fails on this command (newlines added by me):

```
"C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\tools\skshaderc.exe" -o3 -e -t xge 
-i "C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\tools\include" 
-o "C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\bin\x64_Release_UWP\StereoKitTest_UWP\Assets/" 
"C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\Examples\Assets\floor_shader.hlsl"	
```

When I run this command in a dos box to recreate the issue it appears the wrong output file path is used:

```
C:\Users\VR-1>"C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\tools\skshaderc.exe" -o3 -e -t xge -i "C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\tools\include" -o "C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\bin\x64_Release_UWP\StereoKitTest_UWP\Assets/" "C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\Examples\Assets\floor_shader.hlsl"
Compiling C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\Examples\Assets\floor_shader.hlsl..
 ________________
|--Buffer Info--
|  StereoKitBuffer - 784 bytes
|    sk_view        : +0    [  128] - flt32
|    sk_proj        : +128  [  128] - flt32
|    sk_proj_inv    : +256  [  128] - flt32
|    sk_viewproj    : +384  [  128] - flt32
|    sk_lighting_sh : +512  [  144] - flt36
|    sk_camera_pos  : +656  [   32] - flt8
|    sk_camera_dir  : +688  [   32] - flt8
|    sk_fingertip   : +720  [   32] - flt8
|    sk_cubemap_i   : +752  [   16] - flt4
|    sk_time        : +768  [    4] - flt1
|    sk_view_count  : +772  [    4] - uint1
|  TransformBuffer - 65520 bytes
|    sk_inst        : +0    [65520] - misc819
|  $Global - 32 bytes
|    color          : +0    [   16] - flt4
|    radius         : +16   [   16] - flt4
|--Vertex Shader--
|  b1 : StereoKitBuffer
|  b2 : TransformBuffer
|--Pixel Shader--
|  b0 : $Global
|________________

C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\Examples\Assets\floor_shader.hlsl: error: Failed to write file! C:\Users\VR-1\C:\Users\VR-1\Desktop\Projects\StereoKit-paulmelis\bin\x64_Release_UWP\StereoKitTest_UWP\Assets\floor_shader.hlsl.sks


C:\Users\VR-1>
```

Noe the `C:\Users\VR-1\` extra prefix before the output path specified with `-o`. I have no idea why this happens and am not sure it's a local issue with my setup?

**Comment by maluoi** [comment]
Yeah, this looks like an error in skshaderc! I was trying to fix another path related issue, and it looks like I may have introduced another one 😬. I think the core bug in this case is related to the absolute path in `-o`. You can probably work around this for now by using a relative path? I'll look into this, see if I can get a fix in soon.

Actually, are you on an older commit on `/develop`? Just trying this now, it looks like the current command comes out like this:

```
"C:\Data\Repositories\Stereo Kit\tools\skshaderc.exe" -o3 -e -t xge -i "C:\Data\Repositories\Stereo Kit\tools\include" -o "..\..\bin\x64_Release\StereoKitTest\Assets/" "..\Assets\floor_shader.hlsl"
```

**Reply by paulmelis** [reply]
> Actually, are you on an older commit on /develop? 

Hmmm, I just merged in the latest `develop` and I still get the same skshaderc error.

**Reply by paulmelis** [reply]
Indeed, using a relative path with `-o` (and running it separately in a dos box) makes the error go away
